File: CPlusPlus/10845.cpp
#include <bits/stdc++.h>

#define FASTIO                                                                 \
    std::ios_base::sync_with_stdio(false);                                     \
    std::cin.tie(0);                                                           \
    std::cout.tie(0);
#define ENDL "\n";

/***************************************************************************************
 *   * 링크 : https://www.acmicpc.net/problem/10845
 ***************************************************************************************/
using namespace std;

int main() {
    FASTIO

    queue<int> q;
    int TestCase;
    cin >> TestCase;

    for (int i = 0; i < TestCase; i++) {
        string s;
        cin >> s;
        if (s == "push") {
            int temp;
            cin >> temp;
            q.push(temp);
        } else if (s == "pop") {
            if (q.empty()) {
                cout << "-1" << ENDL;
            } else {
                cout << q.front() << ENDL;
                q.pop();
            }
        } else if (s == "size") {
            cout << q.size() << ENDL;
        } else if (s == "empty") {
            if (q.empty()) {
                cout << "1" << ENDL;
            } else {
                cout << "0" << ENDL;
            }
        } else if (s == "front") {
            if (q.empty()) {
                cout << "-1" << ENDL;
            } else {
                cout << q.front() << ENDL;
            }
        } else if (s == "back") {
            if (q.empty()) {
                cout << "-1" << ENDL;
            } else {
                cout << q.back() << ENDL;
            }
        }
    }

    return 0;
}

File: CSharp/10845.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;

// 링크 : https://www.acmicpc.net/problem/10845

namespace BOJ
{
    class Program
    {
        static void Main(string[] args)
        {
            int myBack = 0;
            int testCase = int.Parse(Console.ReadLine());
            Queue<int> q = new Queue<int>();
            StringBuilder sb = new StringBuilder();

            for (int i = 0; i < testCase; i++)
            {
                string[] cmd = Console.ReadLine().Split(' ');

                switch (cmd[0])
                {
                    case "push":
                        myBack = int.Parse(cmd[1]);
                        q.Enqueue(myBack);
                        break;

                    case "pop":
                        sb.AppendLine(q.Count > 0 ? q.Dequeue().ToString() : "-1");
                        break;

                    case "size":
                        sb.Append(q.Count() + "\n");
                        break;
                    case "empty":
                        sb.AppendLine(q.Count == 0 ? "1" : "0");
                        break;
                    case "front":
                        sb.AppendLine(q.Count > 0 ? q.Peek().ToString() : "-1");
                        break;
                    case "back":
                        sb.AppendLine(q.Count > 0 ? myBack.ToString() : "-1");
                        break;
                }
            }

            Console.WriteLine(sb);
        }
    }
}